src/Features/Core/Portable/Snippets/Snippet.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Snippets
+{
+    /// <summary>
+    /// Encapsulates the information that makes up a Snippet.
+    /// </summary>
+    internal readonly struct Snippet
```

Comment:
looking at this, perhaps you should call this SnippetChange to indicate that it's the *change* that needs to be made to apply the snippet to user code.